### PR TITLE
Linux tests + coverage report with Bazel

### PR DESCRIPTION
**Pull request by louwers**
Just trying out Bazel and exploring the build setup.

Two C++ unit tests now can be run with:

```
bazel test //test:core
```

### CODE REVIEW

**Comment by louwers** [comment]
I wanted to use the new [Platforms](https://bazel.build/extending/platforms) functionality to configure which headers include and which flags to pass etc. but it turns out `rules_apple` does not have support for it yet: https://github.com/bazelbuild/rules_apple/issues/1658

I need to find some workaround.

**Comment by louwers** [comment]
coecov.io needs to be enabled, but I don't have organization access. So the upload will fail.

File: platform/linux/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        "-lz",
+        "-lcurl",
+        "-lpng",
+        "-ljpeg",
+    ] + ICU_FLAGS,
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:mbgl-core",
+        "//platform/default:mbgl-default",
+        # "//:headers"
```
Comment (louwers) [inline platform/linux/BUILD.bazel:31]:
```suggestion
```

Comment (louwers) [inline platform/linux/BUILD.bazel:31]:
```suggestion
        # "//:headers"
```

Comment (louwers) [inline platform/linux/BUILD.bazel:31]:
```suggestion
```

**Review by ovivoda — APPROVED** [review]

**Review by mwilsnd — APPROVED** [review]

**Pull request merged**